### ISSUE TITLE
missing cups binding

### ISSUE THREAD

**Issue by staehler** [openhab/openhab1-addons#5859]
The cups binding disappeared from the bindings list, which means I cannot install it. Could you please add it again.
I had problems to get IPP binding running, thats why I'ld prefer to stay with the cups binding.

Im running debian stretch and openhab2 2.5.0~S1587-1.

This [latest cups binding](https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.cups/) doesn't work, as it throws this errors:
```
2019-05-08 21:26:16.386 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/usr/share/openhab2/addons/org.openhab.binding.cups-1.14.0-SNAPSHOT.jar
org.osgi.framework.BundleException: Could not resolve module: org.openhab.binding.cups [200]
2019-05-08 21:26:28.874 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'cups.items'
2019-05-08 21:44:37.538 [WARN ] [org.apache.felix.fileinstall        ] - Error while starting bundle: file:/usr/share/openhab2/addons/org.openhab.binding.cups-1.14.0-SNAPSHOT.jar
org.osgi.framework.BundleException: Could not resolve module: org.openhab.binding.cups [200]
```
Could you please help.

**Comment by kaikreuzer** [comment]
Err, "might not work" should not mean that it isn't at all available anymore...
Is this the only 1.x binding that has disappeared? If so, does anybody know, why this might have happened?

**Comment by staehler** [comment]
Regarding to my addons.cfg I have this OH1 bindings running:
binding=fritzboxtr0641,http1,mqtt1,networkupstools1,plex1,snmp1,weather1,caldav-command1,caldav-personal1,expire1,mqttitude1
It's about 50% of my bindings. Does this really mean, all this binding won't work in the future?
This would be really pitty and could be the end of my openhab installation. Meanwhile I haven't enough time, to spend too much mirgration efforts :-(
It was quite hard to migrate from knx1 to knx2, also the migration from onewire1 to onewire2. The migration of mqtt1 to mqtt2 is on my plan, but loosing all the other bindings is for me a catastroph and means for me either ever never upgrade oh or switch to another home automation platform.

Never the less, I would thank you all very much for all your efforts.

**Comment by J-N-K** [comment]
@staehler Do these bindings work with S1587 and cups is the only missing? Or do they all fail?

**Comment by kaikreuzer** [comment]
> Does this really mean, all this binding won't work in the future?

No, it does not mean this, no worries.

**Comment by kaikreuzer** [comment]
Just checked: All other 1.x bindings are available in the distro.
I actually do not find the cups binding in the [1.x feature list](https://github.com/openhab/openhab1-addons/blob/master/features/openhab-addons/src/main/feature/feature.xml) - it looks to me as if it has never been available in openHAB 2.

@staehler, when you say it has disappeared from the binding list: What was the latest openHAB version, where you saw it as an option?

**Comment by staehler** [comment]
@kaikreuzer, I cannot say, when it disappeared from the bindings list, as it was downloaded automatically, but I first recognized the missing about 20d ago (see https://community.openhab.org/t/cups-binding-not-offered-in-paperui/72826). This workaround helped for the next few days. But now, with the latest openhab version, it doesnt load anymore (see my first post).

That's really great to hear, that the OH1 bindings will work in the future. Thanks Kai for this and all your efforts, I'm a really great openHab fan.

**Comment by staehler** [comment]
@J-N-K yes, the remaining OH1 bindings seem to work. I fight at the moment with some other openhab instabilities, so I couldn't figure out, what is going wrong now. 
1 month ago OH run quite reliable, but with the latest releases lots of unpredictable things happen. squeezebox binding doesn't connect to squeezeserver, KNX blinds sometimes drive, sometimes not, well CUPS disappeared, which explaines that my printer doesn't start.
I'm trying to fall back to an older release and hope that I can get rid of this issues (Milestone 2.5.0M1).

What is the best migration without loosing configurations? 
`
1. openhab-cli backup 
2. uninstall openhab
3. install desiredOH version
4. recover backup`
